nixos/modules/services/web-servers/nginx/location-options.nix

```diff
@@ -111,6 +111,51 @@ with lib;
       '';
     };
 
+    accessLogs = mkOption {
+      type = with types; listOf (submodule { options = {
+        file = mkOption { type = types.nullOr types.path; default = null; description = lib.mdDoc "Set the file for a log write"; };
+        format = mkOption { type = str; default = "combined"; description = lib.mdDoc "Set the format for a log write"; };
+        buffer = mkOption { type = types.nullOr types.str; default = null; description = lib.mdDoc "Set the buffer size for a log write"; };
+        gzip = mkOption { type = types.nullOr types.int; default = null; description = lib.mdDoc "Set the compression level for a log write"; };
+        flush = mkOption { type = types.nullOr types.str; default = null; description = lib.mdDoc "Set the flush time for a log write"; };
+        off = mkOption { type = bool; default = false; description = lib.mdDoc "Cancels all access_log directives on the current level"; };
+      }; });
```

Comment:
We shouldn't duplicate this with vhost if possible

Reply:
Couldn't figure out how to remove duplicates.